### PR TITLE
full august builds

### PR DESCRIPTION
**Pull request by wangzhao0217**
- **add join function**
- **join orcp to netwrok**
- **test it on edinburgh**
- **fix typo**
- **Update with new routes and new date**
- **rev order**
- **updating code**
- **imporve ORCP join**
- **add missing funs**

### CODE REVIEW

**Review by Robinlovelace — COMMENTED** [review]
This looks like great work!

**Comment by Robinlovelace** [comment]
Error message:

```
▶ dispatched target rs_utility_fastest
✖ errored target rs_utility_balanced
✖ errored pipeline [2.727 hours]
Warning message:
6 targets produced warnings. Run targets::tar_meta(fields = warnings, complete_only = TRUE) for the messages. 
Error:
! Error running targets::tar_make()
Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
Debugging guide: https://books.ropensci.org/targets/debugging.html
How to ask for help: https://books.ropensci.org/targets/help.html
Last error message:
    Timeout was reached: [api.cyclestreets.net] Connection timeout after 10001 ms
Last error traceback:
    get_routes(od = od_utility_combined, plans = "balanced", purpose = "util...
    cyclestreets::batch(desire_lines = od, id = id, directory = folder,     ...
    batch_jobdata(username = username, password = password, id = as.characte...
    httr::POST(url = batch_url, body = body)
    request_perform(req, hu$handle$handle)
    request_fetch(req$output, req$url, handle)
    request_fetch.write_memory(req$output, req$url, handle)
    curl::curl_fetch_memory(url, handle = handle)
    .handleSimpleError(function (condition)  {     state$error <- build_mess...
    h(simpleError(msg, call))
> tar_read(parameters)
$plans
[1] "fastest"  "balanced" "quietest" "ebike"   

$min_flow
[1] 1

$max_to_route
[1] 1000000

$date_routing
[1] "2024-09-01"

$geo_subset
[1] FALSE

$open_data_build
[1] FALSE

$disag_threshold
[1] 100

$min_distance_meters
[1] 500

$region_buffer_distance_meters
[1] 5000

$region
[1] "Edinburgh and Lothians"

$coherent_percentile
[1] 0.94
```

**Comment by Robinlovelace** [comment]
Happened again for Glasgow:

```
▶ dispatched target rs_commute_fastest
✖ errored target rs_commute_fastest
✖ errored pipeline [52.908 minutes]
Warning message:
6 targets produced warnings. Run targets::tar_meta(fields = warnings, complete_only = TRUE) for the messages. 
Error:
! Error running targets::tar_make()
Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
Debugging guide: https://books.ropensci.org/targets/debugging.html
How to ask for help: https://books.ropensci.org/targets/help.html
Last error message:
    Timeout was reached: [api.cyclestreets.net] Connection timeout after 10000 ms
Last error traceback:
    get_routes(od = od_commute_subset, plans = "fastest", purpose = "commute...
    cyclestreets::batch(desire_lines = od, id = id, directory = folder,     ...
    batch_jobdata(username = username, password = password, id = as.characte...
    httr::POST(url = batch_url, body = body)
    request_perform(req, hu$handle$handle)
    request_fetch(req$output, req$url, handle)
    request_fetch.write_memory(req$output, req$url, handle)
    curl::curl_fetch_memory(url, handle = handle)
    .handleSimpleError(function (condition)  {     state$error <- build_mess...
    h(simpleError(msg, call))
```

File: route_ids.csv [hunk 1]
```diff
@@ -1,4 +1,11 @@
 nrow,plan,purpose,region,date,id
+218877,fastest,commute,Glasgow and Strathclyde,2024-09-01,10646
```
Comment (Robinlovelace) [inline route_ids.csv:2]: I added this line manually FYI @wangzhao0217

**Comment by Robinlovelace** [comment]
@wangzhao0217 heads-up I think we can fix this by editing httr's upload (POST) timeout limit, could you have a look: https://stackoverflow.com/questions/64367098/specify-timeout-time-in-a-post-or-get-request-with-httr-using-r

**Comment by Robinlovelace** [comment]
Updated upstream, trying again after installing latest version of pkg:

```
remotes::install_dev("cyclestreets")
```

File: code/build.R [hunk 2]
```diff
@@ -200,12 +200,10 @@ for (region in region_names) {
         `Infrastructure type (detailed)` = detailed_segregation,
         `Level of Service`
       )
-    
     # save file for individual district
     district_name = district_geom$LAD23NM |> 
       snakecase::to_snake_case()
     cbd_filename = paste0(output_folder, "/cbd_layer_", district_name, ".geojson")
-
```
Comment (Robinlovelace) [inline code/build.R:205]: Heads-up @wangzhao0217 currently running this for loop.

**Comment by Robinlovelace** [comment]
I just saw this error message, is that expected @wangzhao0217?

```
An error occurred with Na h-Eileanan Siar: argument is of length zero
Generating coherent network for: Highland
Joining with `by = join_by(id)`
[1] "Finished preparing the network data"
```

Is it skipped due to the tryCatch??

**Comment by wangzhao0217** [comment]
No, Robin, the error is because I’m using a fixed version of the OSM data stored on my computer. I’ll do some processing and then publish it

File: code/install.R [hunk 3]
```diff
@@ -12,7 +12,7 @@ if (update_github_packages) {
   remotes::install_github("dabreegster/odjitter", subdir = "r")
   remotes::install_github("robinlovelace/ukboundaries")
   remotes::install_github("robinlovelace/simodels")
-  remotes::install_dev("rsgeo")
+  remotes::install_version("rgeos", version = "0.6-3")
```
Comment (Robinlovelace) [inline code/install.R:15]: Good solution.

**Review by Robinlovelace — COMMENTED** [review]
Looking good, will test now.

**Comment by Robinlovelace** [comment]
Latest on this FYI @wangzhao0217: 

```
Writing 12 features with 2 fields and geometry type Multi Line String.
12 features, 47925 bytes of geometry and attributes, 294 bytes of string pool, 0 bytes of vertices, 0 bytes of nodes
  99.6%  13/4017/2553  % 
Coherent network link for: Edinburgh and Lothians generated successfully
Processing coherent network links for region: Glasgow and Strathclyde
Generating coherent network links for: Glasgow and Strathclyde
Joining with `by = join_by(id)`
[1] "Finished preparing the network data"
```

**Review by Robinlovelace — APPROVED** [review]
Ready to merge when you are @wangzhao0217

**Comment by Robinlovelace** [comment]
Please merge at your leisure...

**Pull request merged**